hibernate-core/src/main/java/org/hibernate/procedure/internal/ProcedureCallImpl.java

```diff
@@ -309,6 +313,8 @@ else if ( parameter.getPosition() != null ) {
 		else {
 			throw new IllegalArgumentException( "Given parameter did not define name or position [" + parameter + "]" );
 		}
+		((ProcedureParameterMetadata)getParameterMetadata()).registerParameter( new ProcedureParameterImpl( parameter ) );
```

Comment:
I suggest an improvement here. This class should override the `getParameterMetadata()` method like this:

```
@Override
public ProcedureParameterMetadata getParameterMetadata() {
    return (ProcedureParameterMetadata) super.getParameterMetadata();
}
```

This way, the line above becomes:

`getParameterMetadata().registerParameter( new ProcedureParameterImpl( registration ) );`

The reason why we want this is because in `ProcedureCallImpl` we should always work with `ProcedureParameterMetadata`, and, this way, we don't need to cast everytime we need the `ProcedureParameterMetadata` type.